kylin/src/main/java/org/apache/zeppelin/kylin/KylinInterpreter.java

```diff
@@ -168,26 +169,31 @@ public String getSQL(String cmd) {
   private InterpreterResult executeQuery(String sql) throws IOException {
 
     HttpResponse response = prepareRequest(sql);
-
-    if (response.getStatusLine().getStatusCode() != 200) {
-      logger.error("failed to execute query: " + response.getEntity().getContent().toString());
-      return new InterpreterResult(InterpreterResult.Code.ERROR,
-          "Failed : HTTP error code " + response.getStatusLine().getStatusCode());
+    String result = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+    int code = response.getStatusLine().getStatusCode();
+    if (code != 200) {
+      StringBuilder errorMessage = new StringBuilder("Failed : HTTP error code " + code);
+      logger.error("failed to execute query: " + result);
+      try {
+        JSONObject content = new JSONObject(result);
+        if (content.has("exception")) {
+          errorMessage.append(". Error message: " + content.getString("exception"));
+        }
+      } catch (JSONException e) {
+        logger.error("Cannot parse json string ", e);
+        // when code is 401, the response is html, not json
+        if (code == 401) {
+          errorMessage.append(". Error message: Unauthorized. This request requires "
+              + "HTTP authentication. Please make sure your have set your credentials correctly.");
+        } else {
+          errorMessage.append(". Error message: " + result);
```

Comment:
previously we were formatting the output (see L186) before setting it to InterpreterResult - do we still need to do that here?

Reply:
@felixcheung yes we still need to do that. The code here is to parse the result and get detailed error message from kylin when query fails. The formatting result will be called when the response code is 200 and query successes.

Reply:
actually, to clarify I was referring to
```
String output;
-    logger.info("Output from Server .... \n");
-    while ((output = br.readLine()) != null) {
-      logger.info(output);
-      sb.append(output).append('\n');
       }
-    InterpreterResult rett = new InterpreterResult(InterpreterResult.Code.SUCCESS, 
-        formatResult(sb.toString()));
```
does it still need to read the result line by line like this from before, before passing to formatResult()?

Reply:
@felixcheung no we do not need to do that. I replace the line by line processing with the following statement:       result = IOUtils.toString(response.getEntity().getContent(), "UTF-8")